libtenzir/builtins/functions/hex.cpp

```diff
@@ -0,0 +1,92 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2024 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "tenzir/arrow_utils.hpp"
+#include "tenzir/tql2/plugin.hpp"
+
+#include <boost/algorithm/hex.hpp>
+
+namespace tenzir::plugins::hex {
+
+namespace {
+
+enum class mode { encode, decode };
+
+template <mode Mode>
+class plugin final : public function_plugin {
+  using Type = std::conditional_t<Mode == mode::encode, string_type, blob_type>;
+
+  auto name() const -> std::string override {
+    return Mode == mode::encode ? "encode_hex" : "decode_hex";
+  }
+
+  auto make_function(invocation inv,
+                     session ctx) const -> failure_or<function_ptr> override {
+    auto expr = ast::expression{};
+    TRY(argument_parser2::function(name())
+          .positional("value", expr, "blob|string")
+          .parse(inv, ctx));
+    return function_use::make([expr = std::move(expr)](evaluator eval,
+                                                       session ctx) -> series {
+      const auto arg = eval(expr);
+      const auto f = detail::overload{
+        [](const arrow::NullArray& array) {
+          return series::null(Type{}, array.length());
+        },
+        [&](const concepts::one_of<arrow::BinaryArray, arrow::StringArray> auto&
+              array) {
+          auto b = Type::make_arrow_builder(arrow::default_memory_pool());
+          check(b->Reserve(array.length()));
+          for (auto i = int64_t{}; i < array.length(); ++i) {
+            if (array.IsNull(i)) {
+              check(b->AppendNull());
+              continue;
+            }
+            const auto val = array.Value(i);
+            auto str = std::string{};
+            if constexpr (Mode == mode::encode) {
+              str.reserve(val.length() * 2);
+              boost::algorithm::hex(val, std::back_inserter(str));
```

Comment:
Why not use `detail::hexify`?

Reply:
Had no idea that existed. Boost already had a simple header for to-fro hex, so I just used it.